### PR TITLE
Jd/emulation problem

### PR DESCRIPTION
**Pull request by jd-lara**
replacement of #716 + other breaking changes. Tests are expected to fail.

### CODE REVIEW

File: src/initial_conditions/initial_condition.jl [hunk 1]
```diff
@@ -1,3 +1,6 @@
+"""
+    Container for the initial condition data
```
Comment (daniel-thom) [inline src/initial_conditions/initial_condition.jl:2]: Tab to the left

File: src/operation/decision_model.jl [hunk 2]
```diff
@@ -320,197 +251,169 @@ function make_time_series_cache(
...
     mkpath(output_dir)
-    set_output_dir!(problem, output_dir)
-    set_console_level!(problem, console_level)
-    set_file_level!(problem, file_level)
+    set_output_dir!(model, output_dir)
+    set_console_level!(model, console_level)
+    set_file_level!(model, file_level)
     TimerOutputs.reset_timer!(BUILD_PROBLEMS_TIMER)
     disable_timer_outputs && TimerOutputs.disable_timer!(BUILD_PROBLEMS_TIMER)
     logger = configure_logging(problem.internal, "w")
```
Comment (daniel-thom) [inline src/operation/decision_model.jl:353]: `problem` is undefined.

File: src/operation/decision_model.jl [hunk 3]
```diff
@@ -584,42 +487,31 @@ results = solve!(OpModel)
...
-    @info "Write results to Store not implemented for problems $T"
+    stats = OptimizerStats(model, step)
+    write_optimizer_stats!(store, get_name(model), stats, start_time)
+    write_model_results!(store, model, start_time; exports = exports)
     return
 end
 
 """
-Default solve method for an operational model used inside of a Simulation. Solves problems that conform to the requirements of OperationsProblem{<: PowerSimulationsOperationsProblem}
+Default solve method for an operational model used inside of a Simulation. Solves problems that conform to the requirements of DecisionModel{<:  DecisionProblem}
```
Comment (daniel-thom) [inline src/operation/decision_model.jl:510]: Nit: extra space

File: src/operation/operation_model_interface.jl [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Default implementations of getter/setter functions for OperationsModel.
```
Comment (daniel-thom) [inline src/operation/operation_model_interface.jl:1]: This file indicates that we would require specific fields for all subtypes of OperationsModel. Are you sure about this?

File: src/simulation/simulation_store.jl [hunk 5]
```diff
@@ -4,26 +4,6 @@ const STORE_CONTAINER_VARIABLES = :variables
 const STORE_CONTAINERS =
     Set((STORE_CONTAINER_DUALS, STORE_CONTAINER_PARAMETERS, STORE_CONTAINER_VARIABLES))
 
-"""
```
Comment (daniel-thom) [inline src/simulation/simulation_store.jl:7]: I'm not sure why you moved this, but the filename now seems odd.

Comment (jd-lara) [reply src/simulation/simulation_store.jl:7]: I moved it to the abstract types definition file to be able to define the DecisionModel before the simulation code.

File: src/operation/operation_model_interface.jl (hunk 4 above)
Comment (jd-lara) [reply src/operation/operation_model_interface.jl:1]: I think we want to have those requirements but probably there is more here than what we want. A lot of these functions need to be eliminated in favor of interfaces directly with the optimization container.

File: src/devices_models/devices/electric_loads.jl [hunk 6]
```diff
@@ -84,21 +83,7 @@ function DeviceRangeConstraintSpec(
     ::Type{<:PM.AbstractPowerModel},
     feedforward::Union{Nothing, AbstractAffectFeedForward},
     use_parameters::Bool,
-    use_forecasts::Bool,
 ) where {T <: PSY.ElectricLoad}
-    if (!use_parameters && !use_forecasts)
```
[the remark this answers is not in the document]
Comment (jd-lara) [reply src/devices_models/devices/electric_loads.jl:89]: we are getting rid of the `use_forecasts` option

**Review by daniel-thom — APPROVED** [review]

**Pull request merged**